x-pack/filebeat/fbreceiver/receiver.go

```diff
@@ -6,28 +6,38 @@ package fbreceiver
 
 import (
 	"context"
+	"fmt"
 	"sync"
 
+	"github.com/elastic/beats/v7/libbeat/api"
 	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/cmd/instance"
+	"github.com/elastic/beats/v7/libbeat/version"
+	"github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/monitoring"
+	metricreport "github.com/elastic/elastic-agent-system-metrics/report"
 
 	"go.opentelemetry.io/collector/component"
 	"go.uber.org/zap"
 )
 
 type filebeatReceiver struct {
-	beat   *beat.Beat
-	beater beat.Beater
-	logger *zap.Logger
-	wg     sync.WaitGroup
+	beat     *instance.Beat
+	beater   beat.Beater
+	logger   *zap.Logger
+	wg       sync.WaitGroup
+	httpConf *config.C
 }
 
 func (fb *filebeatReceiver) Start(ctx context.Context, host component.Host) error {
 	fb.wg.Add(1)
 	go func() {
 		defer fb.wg.Done()
-		fb.logger.Info("starting filebeat receiver")
-		err := fb.beater.Run(fb.beat)
-		if err != nil {
+		if err := fb.startMonitoring(); err != nil {
+			fb.logger.Error("could not start the HTTP server for the API", zap.Error(err))
```

Comment:
I believe we can be a bit more descriptive here: Which HTTP server has failed?
```suggestion
			fb.logger.Error("could not start the HTTP server for the monitoring API", zap.Error(err))
```